pkg/stacker/bom.go

```diff
@@ -0,0 +1,126 @@
+package stacker
+
+import (
+	"fmt"
+	"os"
+
+	"stackerbuild.io/stacker/pkg/container"
+	"stackerbuild.io/stacker/pkg/log"
+	"stackerbuild.io/stacker/pkg/types"
+)
+
+// build for all pkgs and then merge
+func BuildLayerArtifacts(sc types.StackerConfig, storage types.Storage, l types.Layer,
+	tag string, pkg types.Package,
+) error {
+	name, cleanup, err := storage.TemporaryWritableSnapshot(tag)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	c, err := container.New(sc, name)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	err = SetupBuildContainerConfig(sc, storage, c, tag)
+	if err != nil {
+		log.Errorf("build container %v", err)
+		return err
+	}
+
+	err = SetupLayerConfig(sc, c, l, tag)
+	if err != nil {
+		return err
+	}
+
+	binary, err := os.Readlink("/proc/self/exe")
+	if err != nil {
+		return err
+	}
+
+	if err := c.BindMount(binary, "/static-stacker", ""); err != nil {
+		return err
+	}
+
+	cmd := fmt.Sprintf("/static-stacker --oci-dir %s --roots-dir %s --stacker-dir %s --storage-type %s --internal-userns",
+		sc.OCIDir, sc.RootFSDir, sc.StackerDir, sc.StorageType)
+
+	if sc.Debug {
+		cmd += " --debug"
+	}
+
+	cmd += " internal-go"
+
+	author := l.Annotations[types.AuthorAnnotation]
+	org := l.Annotations[types.OrgAnnotation]
+	license := l.Annotations[types.LicenseAnnotation]
+	dest := "/stacker-artifacts"
```

Comment:
can we put this under /stacker ? so /stacker is read-only mount (with imports) and /stacker/artifacts is a read-write mount with this layer's artifacts.  That way we don't pollute the top level any more.